### PR TITLE
Sort up segmentation models

### PR DESCRIPTION
**Pull request by will-jl944**
1. Add BaseSegmenter as the base class for all segmentation models.
2. Add batch inference for segmentation models

### CODE REVIEW

File: paddlex/cv/models/base.py [hunk 1]
```diff
@@ -56,11 +55,15 @@ def net_initialize(self, pretrained_weights=None, save_dir='.'):
                     os.remove(save_dir)
                 os.makedirs(save_dir)
             if self.model_type == 'classifier':
+                if hasattr(self.net, 'scale'):
+                    scale = self.net.scale
+                else:
+                    scale = None
```
Comment (FlyingQianMM) [inline paddlex/cv/models/base.py]: scale = getattr(self.net, 'scale', None)

File: paddlex/cv/models/segmenter.py [hunk 2]
```diff
@@ -253,11 +227,101 @@ def predict(self, img_file, transforms=None):
...
+            im = transforms(im)[0]
+            batch_im.append(im)
+            batch_ori_shape.append(ori_shape)
+        batch_im = paddle.to_tensor(batch_im)
+
+        return batch_im, batch_ori_shape
+
+    @staticmethod
+    def get_batch_restore_list(batch_ori_shape, transforms):
+        batch_restore_list = list()
```
Comment (FlyingQianMM) [inline paddlex/cv/models/segmenter.py:259]: get_batch_restore_list要不要改个名字。record_shape_transform?

File: paddlex/cv/transforms/operators.py [hunk 3]
```diff
@@ -411,6 +415,69 @@ def __call__(self, im, mask=None):
...
 
+class Padding(Transform):
+    def __init__(self,
+                 height,
+                 width,
+                 target_size,
+                 im_padding_value=(127.5, 127.5, 127.5),
+                 label_padding_value=255):
+        if isinstance(target_size, list) or isinstance(target_size, tuple):
+            if len(target_size) != 2:
```
Comment (FlyingQianMM) [inline paddlex/cv/transforms/operators.py:426]: isinstance(target_size, (list, tuple))

File: paddlex/cv/transforms/operators.py [hunk 4]
```diff
@@ -411,6 +415,69 @@ def __call__(self, im, mask=None):
...
+                "Type of target_size is invalid. It should be list or tuple, now is {}"
+                .format(type(target_size)))
+        self.target_h = height
+        self.target_w = width
+        self.target_size = target_size
+        self.im_padding_value = im_padding_value
+        self.label_padding_value = label_padding_value
+
+    def apply_im(self, im):
+        im_height, im_width = im.shape[0], im.shape[1]
```
Comment (FlyingQianMM) [inline paddlex/cv/transforms/operators.py]: im_height, im_width = im.shape[0:2]

File: paddlex/cv/transforms/operators.py [hunk 5]
```diff
@@ -411,6 +415,69 @@ def __call__(self, im, mask=None):
...
+            0,
+            pad_height,
+            0,
+            pad_width,
+            cv2.BORDER_CONSTANT,
+            value=self.im_padding_value)
+        return im
+
+    def apply_mask(self, mask):
+        mask_height, mask_width = mask.shape[0], mask.shape[1]
```
Comment (FlyingQianMM) [inline paddlex/cv/transforms/operators.py]: mask_height, mask_width = mask.shape[0:2]

File: paddlex/cv/models/base.py (hunk 1 above)
Comment (will-jl944) [reply paddlex/cv/models/base.py]: Done

File: paddlex/cv/models/segmenter.py (hunk 2 above)
Comment (will-jl944) [reply paddlex/cv/models/segmenter.py:259]: 已改为get_transforms_shape_info

File: paddlex/cv/transforms/operators.py (hunk 3 above)
Comment (will-jl944) [reply paddlex/cv/transforms/operators.py:426]: Done

File: paddlex/cv/transforms/operators.py (hunk 4 above)
Comment (will-jl944) [reply paddlex/cv/transforms/operators.py]: Done

File: paddlex/cv/transforms/operators.py (hunk 5 above)
Comment (will-jl944) [reply paddlex/cv/transforms/operators.py]: Done

File: paddlex/utils/shm.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+        if len(shm_infos) == 0:
+            return None
+        elif len(shm_infos) == 1:
+            return _parse_size_in_M(shm_infos[0][3])
+        else:
+            shm_infos = [si for si in shm_infos \
+                         if si[-1] == SHM_DEFAULT_MOUNT]
+            if len(shm_infos) == 0:
+                return _parse_size_in_M(shm_infos[0][3])
```
Comment (FlyingQianMM) [inline paddlex/utils/shm.py]: ppdet这里是不是写错了。len(shm_infos)长度为0的话，后面shm_infos[0]会报错

Comment (will-jl944) [reply paddlex/utils/shm.py]: Done

**Pull request merged**